test/Microsoft.NET.Build.Tests/GivenThatWeWantToVerifyNuGetReferenceCompat.cs

```diff
@@ -46,31 +36,53 @@ public void Nuget_reference_compat(string referencerTarget, string testDescripti
                 return;
             }
 
-            foreach (string dependencyTarget in rawDependencyTargets.Split(',', ';', ' ').ToList())
-            {
-                TestProject dependencyProject = GetTestProject(ConstantStringValues.DependencyDirectoryNamePrefix + dependencyTarget.Replace('.', '_'), dependencyTarget, true);
-                TestPackageReference dependencyPackageReference = new(
-                    dependencyProject.Name,
-                    "1.0.0",
-                    ConstantStringValues.ConstructNuGetPackageReferencePath(dependencyProject, identifier: referencerTarget + testDescription + rawDependencyTargets));
+            var dependencyPackageReferences = new List<TestPackageReference>();
 
-                //  Skip creating the NuGet package if not running on Windows; or if the NuGet package already exists
-                //        https://github.com/dotnet/sdk/issues/335
-                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) || dependencyProject.BuildsOnNonWindows)
+            // Process all dependencies in parallel
+            Parallel.ForEach(
```

Comment:
Does this parallelism actually help? It adds a bit of overhead, and we should already be running a bunch of tests in parallel anyway, in which case I imagine all the cores would be in use?

Reply:
Per offline conversation, we don't run the entire test dll in one action but rather run specific test classes. I don't know if dotnet test actually parallelizes those and my experience running locally in the IDE is I don't get parallism. @nohwnd @Evangelink might know better if how we are executing tests is getting any parallelism on helix that would proclude the benefit here but the benefit locally was significant (10m down to 3m)

Reply:
The default for xunit is to parallelize tests within classes, and I don't see any place where you would disable it. Will try it locally.

Reply:
Okay, I was wrong here, unlike mstest and nunit, xunit does not parallelize tests in classes, or in theories. So the tests will run sequentially, there is an issue for it: https://github.com/xunit/xunit/issues/1986 and some workarounds that require customizing the test framework https://www.meziantou.net/parallelize-test-cases-execution-in-xunit.htm

Reply:
Thanks for taking a look. I guess that's a benefit to moving away from xunit if we decide to go there but not a blocker for this PR.

Reply:
Perhaps instead of parallelizing within a single InlineData in a Theory, we could execute each InlineData in parallel? That would be more in line with what the test infrastructure might be able to handle going forward, which means it would be easier to replace if that's added, and it feels more natural to me. Not going to block on that, though.